keanu-project/src/main/java/io/improbable/keanu/util/dot/WriteDot.java

```diff
@@ -0,0 +1,279 @@
+                DotAnnotation vertexAnnotation = v.getClass().getAnnotation(DotAnnotation.class);
+                if (vertexAnnotation != null && vertexAnnotation.displayHyperparameterInfo()) {
+                    applyHyperparameterInfo(v);
+                }
+
+                processedVertices.add(vId);
+            }
+
+            verticesToProcessNow = verticesToProcessNext;
+            iterationIndex++;
+        }
+
+        // Set labels for vertices that have not yet been processed.
+        for (Vertex v : verticesToProcessNow) {
+            setLabel(v);
+        }
+    }
+
+
+    // Utility function for creating a unique identifier for a connection between two vertices.
+    private static String concatenateVertexIds(VertexId v1, VertexId v2) {
+        if (v1.compareTo(v2) < 0) {
+            return v1.toString() + "_" + v2.toString();
+        }
+        else {
+            return v2.toString() + "_" + v1.toString();
+        }
+    }
+
+
+    // Utility function for forming a DOT string to represent an edge.
+    private static String getEdgeString(VertexId v1, VertexId v2) {
+        if (v1.compareTo(v2) < 0) {
+            return  "<" + v1.hashCode() + "> -> <" + v2.hashCode() + ">";
+        }
+        else {
+            return  "<" + v2.hashCode() + "> -> <" + v1.hashCode() + ">";
+        }
+    }
+
+
+    /**
+     * Creates and stores a label for the given vertex.
+     * Label contains the information that will be displayed for this vertex, such as vertex class or display name and value.
+     *
+     * @param v a vertex
+     */
+    private static void setLabel(Vertex v){
+
+        String vertexLabel;
+
+        // If vertex is a constant vertex, display only it's value.
+        if ((vertexLabel = getValueAsString(v)).equals("")) {
+            // Else use vertex label as DOT label for it.
+            DotAnnotation vertexAnnotation = v.getClass().getAnnotation(DotAnnotation.class);
+            if (v.getLabel() != null) {
+                vertexLabel = v.getLabel().getUnqualifiedName();
+            }
+            // If label isn't set, and if DOT annotation is added to this class, use display name for label.
+            else if (vertexAnnotation != null && !vertexAnnotation.displayName().equals("")) {
+                vertexLabel = vertexAnnotation.displayName();
+            }
+            // Else use class name.
+            else {
+                vertexLabel = v.getClass().getSimpleName();
+            }
+        }
+
+        int uniqueHiddenID = v.getId().hashCode();
+        String fullName = uniqueHiddenID + "[label=\"" + vertexLabel + "\"]";
+        idsToLabelStrings.put(uniqueHiddenID, fullName);
+    }
+
+
+    // Get value of constant vertices that always have their value set.
+    private static String getValueAsString(Vertex v) {
+        if (v.getClass().equals(ConstantDoubleVertex.class)) {
+            return "" + ((ConstantDoubleVertex) v).getValue().getValue(0);
```

Comment:
I was under impression that Constant...Vertices store scalar values, but I might have misunderstood. If it's a non-scalar, we probably don't want to display it in a vertex in a graph, as it would get messy, so I should add a check for it. What do you think?

Reply:
Now checking whether the value stored is a scalar and only displaying it if it is,